### PR TITLE
Bdd participant setup

### PR DESCRIPTION
**Pull request by MattMachin**

### CODE REVIEW

File: mobile/tests/e2e/specs/participantSetup.feature [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    Scenario: No unique code generated on subsequent app opening
+        Given: The user has already opened the app and generated a code
+        When: They open the app
+        Then: No unique code will be generated
+        And: The user can interact with the normal app features
+
+    Scenario: Reminder of unique code
+        Given: The user has already opened the app and generated a code
+        And: The user wants a reminder of the unique code
```
Comment (Xiaoxue-Zheng) [inline mobile/tests/e2e/specs/participantSetup.feature:17]: "The user wants" is not a good description

**Comment by Xiaoxue-Zheng** [comment]
When BDD integrate with end 2 end test, each sentence in the BDD will correspond to the operations or element description on the browser, Cypress will support interacting with browsers. 

To better translate BDD to Cypress lauguage(interacting with browsers), I would like to suggest some rules when writing BDD.

Given:
Should contain verbs like: open app/visit url/sign up/sign in/view
One verb in each sentense is prefered.

When:
Should contain verbs like: open/click/select/enter/choose/go back/tap/start
One verb in each sentense is prefered.

Then:
Should contain descriptions like: ticked/opened/visible/displayed(shown)/returned/notified/xxx is correct
Only one element is discribed in one sentense is prefered.
In some cases, these description words may appear in "When" and "Given" sentence, to make sure the steps make sense

File: mobile/tests/e2e/specs/participantSetup.feature [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Story: Create unique code in app
```
[the remark this answers is not in the document]
Comment (MattMachin) [reply mobile/tests/e2e/specs/participantSetup.feature]: I have updated this

File: mobile/tests/e2e/specs/participantSetup.feature [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Story: Create unique code in app 
+    Scenario: Unique code generated on first app opening
+        Given: The user has not previously opened the app
+        When: They open the app
+        Then: A unique code will be generated that is stored in the app
+        And: The user will be shown the code on the screen
+        And: The code will remain on screen until the user dismisses it
```
[the remark this answers is not in the document]
Comment (MattMachin) [reply mobile/tests/e2e/specs/participantSetup.feature]: I have updated this (all occurrences)

File: mobile/tests/e2e/specs/participantSetup.feature [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    Scenario: No unique code generated on subsequent app opening
+        Given: The user has already opened the app and generated a code
+        When: They open the app
+        Then: No unique code will be generated
+        And: The user can interact with the normal app features
```
[the remark this answers is not in the document]
Comment (MattMachin) [reply mobile/tests/e2e/specs/participantSetup.feature]: I have changed this but maybe we need to agree what should be on the homescreen?

File: mobile/tests/e2e/specs/sendResults.feature [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Feature: Send Results
+
+Story: Send results from app to server
```
[the remark this answers is not in the document]
Comment (MattMachin) [reply mobile/tests/e2e/specs/sendResults.feature]: I have updated this

File: mobile/tests/e2e/specs/participantSetup.feature [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+    Scenario: Reminder of unique code
+        Given: The user has already opened the app and generated a code
+        And: The user wants a reminder of the unique code
+        When: They select the "unique code" menu item
```
[the remark this answers is not in the document]
Comment (MattMachin) [reply mobile/tests/e2e/specs/participantSetup.feature]: We should probably discuss this.

File: mobile/tests/e2e/specs/participantSetup.feature (hunk 1 above)
Comment (MattMachin) [reply mobile/tests/e2e/specs/participantSetup.feature]: I have changed this.

File: mobile/tests/e2e/specs/sendResults.feature [hunk 7]
```diff
@@ -0,0 +1,10 @@
+Feature: Send Results
+
+# Actual wording for the message to the user below should come from the study team
+Scenario: Send data on questionnaire completion
+    Given The user has answered some questions on the app
+    When The questions are complete
+    Then A message will be displayed to the user to thank them for completing the questions
+    And The app shall send the data to the server, along with the unique code
+    And The server will store received data for the correct participant
+    And The unique code will be used to identify the correct participant
```
[the remark this answers is not in the document]
Comment (MattMachin) [reply mobile/tests/e2e/specs/sendResults.feature]: Done

File: web/tests/e2e/specs/viewParticipants.feature [hunk 8]
```diff
@@ -0,0 +1,14 @@
+Feature: View participants
+
+Scenario: View participant summary
+    Given The researcher has previously setup one or more participants
+    When The researcher logs in to the web interface
+    Then They will see a "summary view" of existing participants
+    And The view shall be ordered by participant id.
+    And For each participant the view shall include participant id, unique code and a link (e.g. button) to view data
```
[the remark this answers is not in the document]
Comment (MattMachin) [reply web/tests/e2e/specs/viewParticipants.feature]: Done

**Review by Xiaoxue-Zheng — APPROVED** [review]

**Pull request merged**